SpatialConvolution.lua

```diff
@@ -26,14 +26,31 @@ function SpatialConvolution:__init(nInputPlane, nOutputPlane,
     self:reset()
     -- should nil for serialization, the reset will still work
     self.reset = nil
+
+    self._halfPrecision = false
+    devProps = cutorch.getDeviceProperties(cutorch.getDevice())
+    -- half-precision computing (not i/o only) needs HW support (Tegra X1 or Pascal)
```

Comment:
Is this really true? As far as I remember current GPUs can do FP16 computation as well, it's just that they don't have any special HW support for it, so it won't be much faster.